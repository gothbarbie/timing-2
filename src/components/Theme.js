import React from 'react'
import PropTypes from 'prop-types'
import { ThemeProvider } from 'styled-components'

import theme from './themes/default'

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
)

Theme.propTypes = {
  children: PropTypes.node,
}

export default Theme
