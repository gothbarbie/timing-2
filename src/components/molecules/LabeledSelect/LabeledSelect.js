import React from 'react'
import styled from 'styled-components'

import Label from '../../atoms/Label'
import Select from '../../atoms/Select'

const LabeledSelectWrapper = styled.div`
  display: flex;
  flex-direction: column;
`

const LabelStyled = styled(Label)`
  margin-bottom: 1rem;
`

const LabeledSelect = ({ label, name, onChange, options }) => (
  <LabeledSelectWrapper>
    <LabelStyled htmlFor={name}>{label}</LabelStyled>
    <Select name={name} onChange={onChange} options={options} />
  </LabeledSelectWrapper>
)

export default LabeledSelect
