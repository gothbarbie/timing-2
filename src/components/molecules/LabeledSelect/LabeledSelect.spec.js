import React from 'react'
import { shallow } from 'enzyme'

import { theme } from '../../Theme'
import LabeledSelect from './LabeledSelect'

describe('generic/molecules/LabeledSelect', () => {
  it('renders', () => {
    const options = [
      {
        key: 'first',
        value: 1,
      },
    ]
    const component = shallow(
      <LabeledSelect
        theme={theme}
        label="Test"
        onChange={() => jest.fn()}
        name="test"
        options={options}
      />
    )

    expect(component).toMatchSnapshot()
  })
})
