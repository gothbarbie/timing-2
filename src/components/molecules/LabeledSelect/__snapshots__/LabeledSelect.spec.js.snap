// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/molecules/LabeledSelect renders 1`] = `
<styled.div>
  <Styled(styled.label)
    htmlFor="test"
  >
    Test
  </Styled(styled.label)>
  <Select
    name="test"
    onChange={[Function]}
    options={
      Array [
        Object {
          "key": "first",
          "value": 1,
        },
      ]
    }
  />
</styled.div>
`;
